### PR TITLE
Cs events

### PR DESCRIPTION
**Pull request by bpluta-splunk**
### Details

A new detection that uses all the Crowdstrike Events as a risk rule to populate observables. 

### Checklist

- [ ] Validate name matches `<platform>_<mitre att&ck technique>_<short description>` nomenclature
- [ ] [CI/CD](https://github.com/splunk/security_content/actions) jobs passed ✔️ 
- [X ] Validated SPL logic.
- [ X] Validated tags, description, and how to implement.
- [ X] Verified references match analytic.
- [X ] Confirm updates to lookups are handled properly.

### CODE REVIEW

File: detections/endpoint/crowdstrike_event_streams_event.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
+name: CrowdStrike Event Stream Events
```
Comment (patel-bhavin) [inline detections/endpoint/crowdstrike_event_streams_event.yml]: Can we name this detection search better - catering to what the detection is trying to do?

File: detections/endpoint/crowdstrike_event_streams_event.yml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+id: cb6af2b3-29ab-441c-8d8d-679811c8b014
+version: 1
+date: '2025-05-12'
+author: Bryan Pluta, Theoderick Contreras, Splunk
+status: production
+type: Anomaly
+data_source:
+- CS Falcon Event Streams
```
Comment (patel-bhavin) [inline detections/endpoint/crowdstrike_event_streams_event.yml]: Can you provide a data source file  similar to this- https://github.com/splunk/security_content/blob/develop/data_sources/crowdstrike_processrollup2.yml

And consider naming this data_source : Crowdstrike Falcon Event Stream

File: detections/endpoint/crowdstrike_event_streams_event.yml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+description: The following analytic is to leverage alerts from CrowdStrike Falcon Event Stream. This query aggregates and summarizes DetectionSummaryEvent and IdpDetectionSummaryEvent alerts from CrowdStrike Falcon Event Stream, providing details such as destination, user, severity, Mitre information, and Crowdstrike id and links.
```
Comment (patel-bhavin) [inline detections/endpoint/crowdstrike_event_streams_event.yml]: Can you also elaborate on what the evals in the searches  are doing in the description and mention the intention of how this can be useful to surface these CS events as risk/notable objects?

File: detections/endpoint/crowdstrike_event_streams_event.yml [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+  product:
+  - Splunk Enterprise
+  - Splunk Enterprise Security
+  - Splunk Cloud
+  security_domain: endpoint
+  manual_test: We are dynamically creating the risk_score field based on the severity of the alert in the SPL and that supersedes the risk score set in the detection. Setting these to manual test since otherwise we fail integration testing. The detection is also failing on unit-testing as some of the fields set in the observables are empty. 
+tests:
+- name: True Positive Test
+  attack_data:
+  - data: https://media.githubusercontent.com/media/splunk/attack_data/master/datasets/suspicious_behaviour/crowdstrike_stream/event_stream_events/stream_events.log
```
Comment (patel-bhavin) [inline detections/endpoint/crowdstrike_event_streams_event.yml]: @bpluta-splunk  I believe we still need to create a PR for the attack_data

**Comment by patel-bhavin** [comment]
@bpluta-splunk - A few comments on the PR and can you provide us with a screenshot of what this detection output looks like in splunk?

File: detections/endpoint/crowdstrike_event_streams_event.yml [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+drilldown_searches:
+- name: View the detection results for - "$dest$"
+  search: '%original_detection_search% | search  src = "$dest$"'
+  earliest_offset: $info_min_time$
+  latest_offset: $info_max_time$
+- name: View risk events for the last 7 days for - "$dest$"
+  search: '| from datamodel Risk.All_Risk | search normalized_risk_object IN ("$dest$") starthoursago=168  | stats count min(_time) as firstTime max(_time) as lastTime values(search_name) as "Search Name" values(risk_message) as "Risk Message" values(analyticstories) as "Analytic Stories" values(annotations._all) as "Annotations" values(annotations.mitre_attack.mitre_tactic) as "ATT&CK Tactics" by normalized_risk_object | `security_content_ctime(firstTime)` | `security_content_ctime(lastTime)`'
+  earliest_offset: $info_min_time$
+  latest_offset: $info_max_time$
+- name: View other CrowdStrike events for "$user$ on "$dest$"
```
Comment (patel-bhavin) [inline detections/endpoint/crowdstrike_event_streams_event.yml]: Not sure how this works. Can you share an in product screenshot of this drilldown?

File: detections/endpoint/crowdstrike_event_streams_event.yml (hunk 2 above)
Comment (bpluta-splunk) [reply detections/endpoint/crowdstrike_event_streams_event.yml]: I think I tried, but not sure where this commit went to https://github.com/splunk/attack_data/commit/80ac642f59bfbf15f94f820d280ef82be4e0e1e9

File: detections/endpoint/crowdstrike_event_streams_event.yml (hunk 4 above)
Comment (bpluta-splunk) [reply detections/endpoint/crowdstrike_event_streams_event.yml]: Please see the earlier comment about the commit against the attack_data repo

File: detections/endpoint/crowdstrike_event_streams_event.yml (hunk 5 above)
Comment (bpluta-splunk) [reply detections/endpoint/crowdstrike_event_streams_event.yml]: Screenshot 2025-06-25 at 4 07 53 PM
Screenshot 2025-06-25 at 4 08 14 PM
screenshots of the finding and drilldown in splunk

File: detections/endpoint/crowdstrike_event_streams_event.yml (hunk 1 above)
Comment (bpluta-splunk) [reply detections/endpoint/crowdstrike_event_streams_event.yml]: latest commit should address

File: detections/endpoint/crowdstrike_event_streams_event.yml (hunk 3 above)
Comment (bpluta-splunk) [reply detections/endpoint/crowdstrike_event_streams_event.yml]: latest commit should address

File: detections/endpoint/crowdstrike_event_streams_event.yml (hunk 2 above)
Comment (patel-bhavin) [reply detections/endpoint/crowdstrike_event_streams_event.yml]: So this data source file is a security_content/data_sources file and not in attack data. Create a new file similar to https://github.com/splunk/security_content/blob/develop/data_sources/crowdstrike_processrollup2.yml named : Crowdstrike Falcon Event Stream and add details as per the example linked.

File: detections/endpoint/crowdstrike_event_streams_event.yml (hunk 4 above)
Comment (patel-bhavin) [reply detections/endpoint/crowdstrike_event_streams_event.yml]: There was some oddity with the file and git LFS , re uploaded the dataset and it seems to work.

File: detections/endpoint/crowdstrike_falcon_stream_alerts.yml [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
...OT isnull(user)," by " + user,"") | eval dest_append = if(NOT isnull(dest)," on " + dest,"") | eval description = case(DetectName=="NGAV", ("CS " + Tactic + " - " + Technique + ": " + FileName),eventType=="IdpDetectionSummaryEvent", ("CS IdP" + " - " + DetectName),DetectName=="Intel Detection", ("CS " + DetectName + " - " + IOCType + ": " + IOCValue),1==1, (Objective + " - " + DetectDescription) ) | eval description = description + user_append + dest_append | eval gid=DetectId, display_id=FalconHostLink, file_hash=SHA256String, hash=MD5String, signature=IOCValue, ip=''NetworkAccesses{}.RemoteAddress'', process=CommandLine, pid=ProcessId  | eval file_name = if(isnull(''ExecutablesWritten{}.FileName''),FileName,''ExecutablesWritten{}.FileName'') | rename DetectId as detection_id, FalconHostLink as detection_url `crowdstrike_event_stream_events_filter` | table _time source detection_id detection_url title risk_score description Severity severity ComputerName dest Tactic Technique user UserName SourceAccountName Objective DetectName DetectDescription gid, display_id, mitre_technique annotations.mitre_attack annotations.drilldown_search file_hash hash signature ip process pid file_name'
```
Comment (patel-bhavin) [inline detections/endpoint/crowdstrike_falcon_stream_alerts.yml]: Lets remove `index=falcon_edr` from the detection logic

File: detections/endpoint/crowdstrike_event_streams_event.yml (hunk 5 above)
Comment (patel-bhavin) [reply detections/endpoint/crowdstrike_event_streams_event.yml]: great, where can we see the list of drill downs on the first screenshot?

**Comment by patel-bhavin** [comment]
@bryan-splunk - we highly recommend you check the CI output of the build phase to work through the CI errors - 


- the filter macro in the SPL needs an update 

- Looks like the search string is not correctly formatted. 

https://github.com/splunk/security_content/actions/runs/15887420018/job/44802697607?pr=3544

File: detections/endpoint/crowdstrike_falcon_stream_alerts.yml [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+how_to_implement: In order to properly run this search, you need to ingest alerts data from CrowdStrike Event Stream, specifcally using the CrowdStrike Falcon Event Streams Technical Add-On. This add-on will collect alerts using the CrowdStrike:Event:Streams:JSON sourcetype. You will need to define the `crowdstrike_stream` macro to point to the proper index that contains the CrowdStrike:Event:Streams:JSON sourcetype.
+known_false_positives: False positives may vary based on Crowdstrike configuration; monitor and filter out the alerts that are not relevant to your environment.
+references:
+- https://www.crowdstrike.com/en-us/resources/guides/crowdstrike-falcon-event-streams-add-on-for-splunk-guide-v3/
+- https://splunkbase.splunk.com/app/5082
+drilldown_searches:
+- name: View the detection results for - "$dest$"
+  search: '%original_detection_search% | search  src = "$dest$"'
```
Comment (patel-bhavin) [inline detections/endpoint/crowdstrike_falcon_stream_alerts.yml]: should be dest = "$dest$"

File: detections/endpoint/crowdstrike_event_streams_event.yml (hunk 2 above)
Comment (bpluta-splunk) [reply detections/endpoint/crowdstrike_event_streams_event.yml]: If I follow something like https://github.com/splunk/security_content/blob/develop/detections/endpoint/crowdstrike_medium_severity_alert.yml then do I even need to specify a data_source?  What is the reasoning of having data_source specified in the detection yaml?

File: detections/endpoint/crowdstrike_event_streams_event.yml (hunk 5 above)
Comment (bpluta-splunk) [reply detections/endpoint/crowdstrike_event_streams_event.yml]: The right side panel was not scrolled down. This show the drilldown search
Screenshot 2025-06-26 at 8 00 07 AM

File: detections/endpoint/crowdstrike_falcon_stream_alerts.yml (hunk 7 above)
Comment (bpluta-splunk) [reply detections/endpoint/crowdstrike_falcon_stream_alerts.yml]: the risk_message field is not going to be defined as part of the search because it is a built in field that can dynamically be assigned to.

Comment (patel-bhavin) [reply detections/endpoint/crowdstrike_falcon_stream_alerts.yml]: I pushed an update to the PR to update the formatting of the search key and removed $risk_message$ from the risk_message field. The spl as is today doesnt seem to be generating a risk_message field, maybe we have missed that logic. 

The build should now pass the CI and waiting on unit-testing results

**Comment by patel-bhavin** [comment]
@bryan-splunk - testing using the attack data, made some fixes to the eval since they had some issues

 also commented the input lookup out for unit-testing purposes (will revert once CI run is fine and will add manual test")

You can check the fixes to SPL here : https://github.com/splunk/security_content/pull/3544/commits/983b3b92631c9e5081e6c840bdab106afef4b5f9

Testing screenshot from local machine.
image

File: detections/endpoint/crowdstrike_falcon_stream_alerts.yml (hunk 7 above)
Comment (bpluta-splunk) [reply detections/endpoint/crowdstrike_falcon_stream_alerts.yml]: If we have a way to use what is in the description field as the risk_message that is what we should do. Some of this was me trying to test and generating a finding, which uses the description field. I would think under normal conditions, this would be a risk rule, or an intermediate finding meaning the risk_message field would be the same as what is now the description field.

**Comment by patel-bhavin** [comment]
the latest commit message is misleading - `[remove detections](https://github.com/splunk/security_content/pull/3544/commits/cdfc690acdcf110383c21a500ae0ca599e99613e)`

the latest commit adding manual check back to the PR

File: detections/endpoint/crowdstrike_event_streams_event.yml (hunk 2 above)
Comment (patel-bhavin) [reply detections/endpoint/crowdstrike_event_streams_event.yml]: yes, we are now enforcing having this data_source object as it helps with automatic data source check on the ESCU  Analytic Story onboarding assistant. 

It also specifically helps with understanding what the underlying dataset of a said detection looks like . I have added one to this PR!

File: detections/endpoint/crowdstrike_falcon_stream_alerts.yml (hunk 7 above)
Comment (patel-bhavin) [reply detections/endpoint/crowdstrike_falcon_stream_alerts.yml]: yes , I added " $descirption$ " to the message field and leaving this as a Anomaly which means that this will generate a risk event.

**Review by patel-bhavin — APPROVED** [review]

**Review by patel-bhavin — COMMENTED** [review]
@bryan-splunk - Can you perhaps take a look at the updated PR ? I added $description$ to the risk_message as per our discussion. Is that sufficient?

**Comment by bpluta-splunk** [comment]
By using the description in the message field for RBA, that will make it is a customer wants to switch this from an intermediate finding to a finding, everything should still work with no additional changes. Everything is looking good. Please let me know if there is something else that is needed for this or anything else that need to be changed or reviewed.
